src/Common/Core/Header/Asset.php

```diff
@@ -26,6 +26,17 @@ public function __construct(string $file, bool $addTimestamp = false, Priority $
         $this->createdOn = new DateTimeImmutable();
     }
 
+    public function compare(Asset $asset)
+    {
+        $comparison = $this->priority->compare($asset->getPriority());
+
+        if ($comparison === 0) {
+            $comparison = $this->createdOn <=> $asset->getCreatedOn();
```

Comment:
since it is the same class you can remove the getter and just use `$asset->createdOn` because it is the same class you can access the private properties